eZ/Bundle/EzPublishRestBundle/Tests/EventListener/RequestListenerTest.php

```diff
@@ -19,42 +19,106 @@
+    const NON_REST_ROUTE = '/non-rest-route';
 
     public function provideExpectedSubscribedEventTypes()
     {
-        return array(
-            array(array(KernelEvents::REQUEST)),
-        );
+        return [
+            [
+                [KernelEvents::REQUEST],
+            ],
+        ];
     }
 
     public function testOnKernelRequestNotMasterRequest()
     {
-        $event = $this->getEvent(self::REST_PREFIX . '/', HttpKernelInterface::SUB_REQUEST);
-        $this->getEventListener()->onKernelRequest($event);
+        $request = $this->performFakeRequest(self::REST_ROUTE, HttpKernelInterface::SUB_REQUEST);
 
-        self::assertTrue(
-            $event->getRequest()->attributes->get('is_rest_request')
-        );
+        self::assertTrue($request->attributes->get('is_rest_request'));
     }
 
     public function testOnKernelRequestNotRestRequest()
     {
-        $event = $this->getEvent('/');
-        $this->getEventListener()->onKernelRequest($event);
+        $request = $this->performFakeRequest(self::NON_REST_ROUTE);
 
-        self::assertFalse(
-            $event->getRequest()->attributes->get('is_rest_request')
-        );
+        self::assertFalse($request->attributes->get('is_rest_request'));
     }
 
     public function testOnKernelRequestRestRequest()
     {
-        $event = $this->getEvent(self::REST_PREFIX . '/');
-        $this->getEventListener()->onKernelRequest($event);
+        $request = $this->performFakeRequest(self::REST_ROUTE);
+
+        self::assertTrue($request->attributes->get('is_rest_request'));
+    }
+
+    public function testRestRequestVariations()
+    {
+        self::assertTrue(
+            $this->performFakeRequest('/api/ezp/v2/true')->attributes->get('is_rest_request')
+        );
 
         self::assertTrue(
-            $event->getRequest()->attributes->get('is_rest_request')
+            $this->performFakeRequest('/api/bundle-name/v2/true')->attributes->get('is_rest_request')
+        );
+
+        self::assertTrue(
+            $this->performFakeRequest('/api/MyBundle12/v2/true')->attributes->get('is_rest_request')
+        );
+
+        self::assertTrue(
+            $this->performFakeRequest('/api/ThisIs_Bundle123/v2/true')->attributes->get('is_rest_request')
+        );
+
+        self::assertTrue(
+            $this->performFakeRequest('/api/my-bundle/v1/true')->attributes->get('is_rest_request')
+        );
+
+        self::assertTrue(
+            $this->performFakeRequest('/api/my-bundle/v2/true')->attributes->get('is_rest_request')
+        );
+
+        self::assertTrue(
+            $this->performFakeRequest('/api/my-bundle/v2.7/true')->attributes->get('is_rest_request')
+        );
+
+        self::assertTrue(
+            $this->performFakeRequest('/api/my-bundle/v122.73/true')->attributes->get('is_rest_request')
+        );
+    }
+
+    public function testNonRestRequestVariations()
+    {
+        self::assertFalse(
```

Comment:
It won't make much of a difference, but why did you go for this iterative approach instead of a dataProvider ?